test/api/api_test.go

```diff
@@ -154,106 +298,98 @@ func TestUserTierCache(t *testing.T) {
+	dep := dependencies.NewDependencyMongoWriteConflictN(api.DBTxnRetryCount + 1)
+	at, err := test.NewAccountsTester(dbName, dep)
 	if err != nil {
 		t.Fatal(err)
 	}
-	// We need to try this several times because we'll only get the right result
-	// after the background goroutine that updates user's quotas has had time to
-	// run.
-	err = build.Retry(10, 200*time.Millisecond, func() error {
-		// We expect to get tier with name and id matching TierPremium20 but with
-		// speeds matching TierAnonymous.
-		ul, _, err = at.UserLimits("byte", nil)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if ul.TierID != database.TierPremium20 {
-			return fmt.Errorf("Expected tier id '%d', got '%d'", database.TierPremium20, ul.TierID)
-		}
-		if ul.TierName != database.UserLimits[database.TierPremium20].TierName {
-			return fmt.Errorf("Expected tier name '%s', got '%s'", database.UserLimits[database.TierPremium20].TierName, ul.TierName)
-		}
-		if ul.UploadBandwidth != database.UserLimits[database.TierAnonymous].UploadBandwidth {
-			return fmt.Errorf("Expected upload bandwidth '%d', got '%d'", database.UserLimits[database.TierAnonymous].UploadBandwidth, ul.UploadBandwidth)
+	defer func() {
+		if errClose := at.Close(); errClose != nil {
+			t.Error(errors.AddContext(errClose, "failed to close account tester"))
 		}
-		return nil
-	})
+	}()
+
+	// Create a test user.
+	userEmailStr := types.NewEmail(t.Name() + "@siasky.net").String()
+	userPassword := t.Name() + "pass"
+	_, b, err := at.UserPOST(userEmailStr, userPassword)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal(err, string(b))
 	}
-	// Delete the uploaded file, so the user's quota recovers.
-	// This call should invalidate the tier cache.
-	_, err = at.UploadsDELETE(sl.Skylink)
+	defer func() {
+		_, _ = at.UserDELETE()
+	}()
+	r, b, err := at.LoginCredentialsPOST(userEmailStr, userPassword)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatal(err, string(b))
 	}
-	err = build.Retry(10, 200*time.Millisecond, func() error {
-		// We expect to get TierPremium20.
-		ul, _, err = at.UserLimits("byte", nil)
-		if err != nil {
-			return errors.AddContext(err, "failed to call /user/limits")
-		}
-		if ul.TierID != database.TierPremium20 {
-			return fmt.Errorf("Expected tier id '%d', got '%d'", database.TierPremium20, ul.TierID)
-		}
-		if ul.TierName != database.UserLimits[database.TierPremium20].TierName {
-			return fmt.Errorf("Expected tier name '%s', got '%s'", database.UserLimits[database.TierPremium20].TierName, ul.TierName)
-		}
-		return nil
-	})
+	at.SetCookie(test.ExtractCookie(r))
+
+	// Try to update the user. Expect this to fail due to the dependency.
+	_, _, err = at.UserPUT(userEmailStr, "new password", "")
+	if err == nil || !strings.Contains(err.Error(), dependencies.DependencyMongoWriteConflictNMessage) {
+		t.Fatalf("Expected a '%s' error, got '%s'", dependencies.DependencyMongoWriteConflictNMessage, err)
+	}
+	// Try to update the user again. Expect this to pass after one retry.
```

Comment:
should we assert the PUT was successful? so check whether the password effectively updated?